clang/lib/Driver/Driver.cpp

```diff
@@ -3743,6 +3743,16 @@ class OffloadingActionBuilder final {
           for (auto SDA : SYCLDeviceActions)
             SYCLLinkBinaryList.push_back(SDA);
           if (WrapDeviceOnlyBinary) {
+            // If used without -fintelfpga, -fsycl-link is used to wrap device
+            // objects for future host link. Device libraries should be linked
+            // by default to resolve any undefined reference.
+            if (!Args.hasArg(options::OPT_fintelfpga)) {
+              const auto *TC = ToolChains.front();
+              addSYCLDeviceLibs(TC, SYCLLinkBinaryList, true,
+                                C.getDefaultToolChain()
+                                    .getTriple()
+                                    .isWindowsMSVCEnvironment());
+            }
```

Comment:
There is a current issue where usage of `-fintelfpga` is too liberal, where we should be using the `spir64_fpga` triple instead.  Should that be the case here?  Downside would be that regular `-fsycl-link` usage would not be possible with FPGA and usage of `-fsycl-link` would be only to create FPGA specific device libraries.

Reply:
Hi, @mdtoguchi 
According to latest user manual, -fsycl-link has 2 different user scenario which depends on whether it is used together with -fintelfpga. In this PR, we only want to toch the "pure" fsycl-link mode which is used to create a device object which is later used in host link.
Do you mean compiler driver will change the behavior of "-fsycl-link", it will only be used to create FPGA specific device libraries? If so, what option/command should we use to create device object used for host link?
Thanks very much.

Reply:
Use of `-fsycl-link` is not changing.  What I mean is that use of `-fintelfpga` and `-fsycl-targets=spir64_fpga-unknown-unknown-sycldevice` should be equivalent sans a couple implied defaults with `-fintelfpga`.

Reply:
So, I need to have a more solid check for FPGA in the patch? Checking -fintelfpga seems not to be enough, we need to check whether spir64_fpga triple is specified too. Did I get your point correctly?
Thanks very much.

Reply:
I would say just checking for SPIRSubArch_fpga should be sufficient, as `-fintelfpga` implies it.

Reply:
Hi, @mdtoguchi 
If -fsycl-link is used together with -fintelfpga, sycl device libraries will be added via: https://github.com/intel/llvm/blob/sycl/clang/lib/Driver/Driver.cpp#L4094
in the function appendLinkDependences.
If -fsycl-link is used without -fintelfpga, the code path is different, we will step into getDeviceDependences. I am not sure whether we have real world scenario for the combination: "-fsycl-link -fsycl-targets=spir64_fpga-unknown-unknwon-sycldevice", I tried this combination and found the we still need to add sycl device libraries in getDeviceDependences.
In a word, although -fintelfpga and -fsycl-targets=spir64_fpga-unknown-unknown-sycldevice may be equivalent, we have different code paths in driver. I suggest to keep the check for fintelfpga here.
Thanks very much.

Reply:
Okay.  You can continue to use `-fintelfpga` for this path, as the behavioral trigger is currently `-fsycl-link -fintelfpga`.  We may need to weigh in on `-fsycl-link` behaviors with `-fsycl-targets=spir64_fpga` as we look to keed a differentiator other than use of the FPGA triple.  Thanks